Sources/MapboxMaps/Foundation/DictionaryCoding/DictionaryEncoder.swift

```diff
@@ -0,0 +1,399 @@
+import Foundation
+
+private protocol EncoderContainer {
+    func toAny() throws -> Any
+}
+
+internal final class DictionaryEncoder {
+    var userInfo: [CodingUserInfoKey: Any] = [:]
+    init() {}
+
+    func encode<T>(_ value: T) throws -> [String: Any] where T: Encodable {
+        guard let dictionary = try Encoder(userInfo: userInfo).encode(value) as? [String: Any] else {
+            throw Error.unexpectedType
+        }
+
+        return dictionary
+    }
+
+    enum Error: Swift.Error {
+        case unexpectedType
+        case incomplete(at: [CodingKey])
+    }
+
+    private static func isSupportedType<T: Encodable>(_ type: T.Type) -> Bool {
+        T.self == Data.self || T.self == NSData.self ||
+        T.self == Date.self || T.self == NSDate.self ||
+        T.self == Decimal.self || T.self == NSDecimalNumber.self ||
+        T.self == URL.self || T.self == NSURL.self
```

Comment:
What does it mean that only these types are supported?

Reply:
aye, the function name should be changed into something that makes more sense, but this is to check for the type and encode them properly (maybe later will be based on configuration i.e data will be encoded with base64, date will be encoded with ISO8601 etc)
I will rename the function.